components/compositing/compositor.rs

```diff
@@ -165,7 +165,7 @@ pub struct IOCompositor<Window: WindowMethods + ?Sized> {
     webrender_api: RenderApi,
 
     /// The surfman instance that webrender targets
-    rendering_context: RenderingContext,
+    rendering_context: R,
```

Comment:
Instead of adding a new parameter to the structure, I think this should hold either `Box<dyn RenderingContext>` or `Rc<dyn RenderingContext>`.

Reply:
Actually I changed it to generics because @wusyong asked me to, I was wondering about it too after seeing #35121

Reply:
@dklassic Apologies! I didn't notice this. Yes. We are starting to move away from making the `Compositor` generic on these types. Hopefully that is okay with you @wusyong.

Sorry for making you do and undo something.

Reply:
@mrobinson  What's the reason to move away generic? Shouldn't generic have less overhead?